### ISSUE TITLE
ReLU6 support

### ISSUE THREAD

**Issue by ankandrew** [AlexeyAB/darknet#6069]
Is it possible to add ReLU6 as an activation function, so you can change layers activations to `activation=relu6` in .cfg?
I am interested in this because I want to change yolov4 tiny with relu6 for later quantization for coral edge tpu
Thanks)

**Comment by AlexeyAB** [comment]
Use `activation=relu6`

But there is no pseudo-quantization.

RELU6 is added, but isn't tested well:
* https://github.com/AlexeyAB/darknet/blob/a71b9a6e9a009cf94900c53deb344c5204835700/src/activation_kernels.cu#L32
* https://github.com/AlexeyAB/darknet/blob/a71b9a6e9a009cf94900c53deb344c5204835700/src/activation_kernels.cu#L73
* https://github.com/AlexeyAB/darknet/blob/a71b9a6e9a009cf94900c53deb344c5204835700/src/activations.c#L58

**Comment by ankandrew** [comment]
> But there is no pseudo-quantization.

What does pseudo-quantization mean? My idea was to [export](https://github.com/hunglc007/tensorflow-yolov4-tflite) to TF and then do post-training quantization (I think it is possible).



> RELU6 is added, but isn't tested well:

I am currently training yolo v4 tiny (license plates) I will experiment with default .cfg (leaky) and then try relu6. I will post the results here.

**Comment by ankandrew** [comment]
@AlexeyAB I changed the .cfg activations with relu6 an re-trained, and I see very similar results. I am a little confuse if it is possible to change the activation function, since the pre-trained weights were trained with leaky and not with relu6.

### relu6
![chart_yolov4-tiny-relu6](https://user-images.githubusercontent.com/61120139/85879233-5d8efb00-b7b0-11ea-933a-3899c9be054f.png)


<details> 
	<summary><b>relu6 cfg:</b></summary> 

	[net]
	# Testing
	#batch=1
	#subdivisions=1
	# Training
	batch=64
	subdivisions=8
	width=416
	height=416
	channels=3
	momentum=0.9
	decay=0.0005
	angle=0
	saturation = 1.5
	exposure = 1.5
	hue=.1

	learning_rate=0.00261
	burn_in=1000
	max_batches = 6000
	policy=steps
	steps=4800,5400
	scales=.1,.1

	mosaic=1
	mosaic_bound=1

	[convolutional]
	batch_normalize=1
	filters=32
	size=3
	stride=2
	pad=1
	activation=relu6

	[convolutional]
	batch_normalize=1
	filters=64
	size=3
	stride=2
	pad=1
	activation=relu6

	[convolutional]
	batch_normalize=1
	filters=64
	size=3
	stride=1
	pad=1
	activation=relu6

	[route]
	layers=-1
	groups=2
	group_id=1

	[convolutional]
	batch_normalize=1
	filters=32
	size=3
	stride=1
	pad=1
	activation=relu6

	[convolutional]
	batch_normalize=1
	filters=32
	size=3
	stride=1
	pad=1
	activation=relu6

	[route]
	layers = -1,-2

	[convolutional]
	batch_normalize=1
	filters=64
	size=1
	stride=1
	pad=1
	activation=relu6

	[route]
	layers = -6,-1

	[maxpool]
	size=2
	stride=2

	[convolutional]
	batch_normalize=1
	filters=128
	size=3
	stride=1
	pad=1
	activation=relu6

	[route]
	layers=-1
	groups=2
	group_id=1

	[convolutional]
	batch_normalize=1
	filters=64
	size=3
	stride=1
	pad=1
	activation=relu6

	[convolutional]
	batch_normalize=1
	filters=64
	size=3
	stride=1
	pad=1
	activation=relu6

	[route]
	layers = -1,-2

	[convolutional]
	batch_normalize=1
	filters=128
	size=1
	stride=1
	pad=1
	activation=relu6

	[route]
	layers = -6,-1

	[maxpool]
	size=2
	stride=2

	[convolutional]
	batch_normalize=1
	filters=256
	size=3
	stride=1
	pad=1
	activation=relu6

	[route]
	layers=-1
	groups=2
	group_id=1

	[convolutional]
	batch_normalize=1
	filters=128
	size=3
	stride=1
	pad=1
	activation=relu6

	[convolutional]
	batch_normalize=1
	filters=128
	size=3
	stride=1
	pad=1
	activation=relu6

	[route]
	layers = -1,-2

	[convolutional]
	batch_normalize=1
	filters=256
	size=1
	stride=1
	pad=1
	activation=relu6

	[route]
	layers = -6,-1

	[maxpool]
	size=2
	stride=2

	[convolutional]
	batch_normalize=1
	filters=512
	size=3
	stride=1
	pad=1
	activation=relu6

	##################################

	[convolutional]
	batch_normalize=1
	filters=256
	size=1
	stride=1
	pad=1
	activation=relu6

	[convolutional]
	batch_normalize=1
	filters=512
	size=3
	stride=1
	pad=1
	activation=relu6

	[convolutional]
	size=1
	stride=1
	pad=1
	filters=18
	activation=linear



	[yolo]
	mask = 3,4,5
	anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
	classes=1
	num=6
	jitter=.3
	scale_x_y = 1.05
	cls_normalizer=1.0
	iou_normalizer=0.07
	iou_loss=ciou
	ignore_thresh = .7
	truth_thresh = 1
	random=0
	resize=1.5
	nms_kind=greedynms
	beta_nms=0.6

	[route]
	layers = -4

	[convolutional]
	batch_normalize=1
	filters=128
	size=1
	stride=1
	pad=1
	activation=relu6

	[upsample]
	stride=2

	[route]
	layers = -1, 23

	[convolutional]
	batch_normalize=1
	filters=256
	size=3
	stride=1
	pad=1
	activation=relu6

	[convolutional]
	size=1
	stride=1
	pad=1
	filters=18
	activation=linear

	[yolo]
	mask = 1,2,3
	anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
	classes=1
	num=6
	jitter=.3
	scale_x_y = 1.05
	cls_normalizer=1.0
	iou_normalizer=0.07
	iou_loss=ciou
	ignore_thresh = .7
	truth_thresh = 1
	random=0
	resize=1.5
	nms_kind=greedynms
	beta_nms=0.6

</details>  




### default(leaky)
![chart_yolov4-tiny-8](https://user-images.githubusercontent.com/61120139/85879235-5ec02800-b7b0-11ea-8014-5577497cea9e.png)

<details> 
	<summary><b>leaky cfg</b></summary>
	
	[net]
	# Testing
	batch=1
	subdivisions=1
	# Training
	#batch=64
	#subdivisions=8
	width=416
	height=416
	channels=3
	momentum=0.9
	decay=0.0005
	angle=0
	saturation = 1.5
	exposure = 1.5
	hue=.1

	learning_rate=0.00261
	burn_in=1000
	max_batches = 6000
	policy=steps
	steps=4800,5400
	scales=.1,.1

	mosaic=1
	mosaic_bound=1

	[convolutional]
	batch_normalize=1
	filters=32
	size=3
	stride=2
	pad=1
	activation=leaky

	[convolutional]
	batch_normalize=1
	filters=64
	size=3
	stride=2
	pad=1
	activation=leaky

	[convolutional]
	batch_normalize=1
	filters=64
	size=3
	stride=1
	pad=1
	activation=leaky

	[route]
	layers=-1
	groups=2
	group_id=1

	[convolutional]
	batch_normalize=1
	filters=32
	size=3
	stride=1
	pad=1
	activation=leaky

	[convolutional]
	batch_normalize=1
	filters=32
	size=3
	stride=1
	pad=1
	activation=leaky

	[route]
	layers = -1,-2

	[convolutional]
	batch_normalize=1
	filters=64
	size=1
	stride=1
	pad=1
	activation=leaky

	[route]
	layers = -6,-1

	[maxpool]
	size=2
	stride=2

	[convolutional]
	batch_normalize=1
	filters=128
	size=3
	stride=1
	pad=1
	activation=leaky

	[route]
	layers=-1
	groups=2
	group_id=1

	[convolutional]
	batch_normalize=1
	filters=64
	size=3
	stride=1
	pad=1
	activation=leaky

	[convolutional]
	batch_normalize=1
	filters=64
	size=3
	stride=1
	pad=1
	activation=leaky

	[route]
	layers = -1,-2

	[convolutional]
	batch_normalize=1
	filters=128
	size=1
	stride=1
	pad=1
	activation=leaky

	[route]
	layers = -6,-1

	[maxpool]
	size=2
	stride=2

	[convolutional]
	batch_normalize=1
	filters=256
	size=3
	stride=1
	pad=1
	activation=leaky

	[route]
	layers=-1
	groups=2
	group_id=1

	[convolutional]
	batch_normalize=1
	filters=128
	size=3
	stride=1
	pad=1
	activation=leaky

	[convolutional]
	batch_normalize=1
	filters=128
	size=3
	stride=1
	pad=1
	activation=leaky

	[route]
	layers = -1,-2

	[convolutional]
	batch_normalize=1
	filters=256
	size=1
	stride=1
	pad=1
	activation=leaky

	[route]
	layers = -6,-1

	[maxpool]
	size=2
	stride=2

	[convolutional]
	batch_normalize=1
	filters=512
	size=3
	stride=1
	pad=1
	activation=leaky

	##################################

	[convolutional]
	batch_normalize=1
	filters=256
	size=1
	stride=1
	pad=1
	activation=leaky

	[convolutional]
	batch_normalize=1
	filters=512
	size=3
	stride=1
	pad=1
	activation=leaky

	[convolutional]
	size=1
	stride=1
	pad=1
	filters=18
	activation=linear



	[yolo]
	mask = 3,4,5
	anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
	classes=1
	num=6
	jitter=.3
	scale_x_y = 1.05
	cls_normalizer=1.0
	iou_normalizer=0.07
	iou_loss=ciou
	ignore_thresh = .7
	truth_thresh = 1
	random=0
	resize=1.5
	nms_kind=greedynms
	beta_nms=0.6

	[route]
	layers = -4

	[convolutional]
	batch_normalize=1
	filters=128
	size=1
	stride=1
	pad=1
	activation=leaky

	[upsample]
	stride=2

	[route]
	layers = -1, 23

	[convolutional]
	batch_normalize=1
	filters=256
	size=3
	stride=1
	pad=1
	activation=leaky

	[convolutional]
	size=1
	stride=1
	pad=1
	filters=18
	activation=linear

	[yolo]
	mask = 1,2,3
	anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
	classes=1
	num=6
	jitter=.3
	scale_x_y = 1.05
	cls_normalizer=1.0
	iou_normalizer=0.07
	iou_loss=ciou
	ignore_thresh = .7
	truth_thresh = 1
	random=0
	resize=1.5
	nms_kind=greedynms
	beta_nms=0.6
</details>  

Should I also try removing mosaic augmentation?
